s3tests_boto3/functional/test_iam.py

```diff
@@ -1591,3 +1593,171 @@ def test_deny_version_id_condition_in_user_policy():
+    client = get_iam_client()
+    # Create bucket user1buck and upload object srcobj as user1
+    s3_client_iam = get_iam_s3client()
+    user1buck = get_new_bucket(client=s3_client_iam)
+    s3_client_iam.put_object(Bucket=user1buck, Key='srcobj', Body='bar')
+    # Create bucket user2buck as user2
+    s3_client_alt = get_alt_client()
+    user2buck = get_new_bucket(client=s3_client_alt)
+    # Add inline user policy with Deny s3:GetObject on user1buck for user2
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Deny",
+             "Action": ["s3:GetObject"],
+             "Resource": f"arn:aws:s3:::{user1buck}/*"}}
+    )
+    response = client.put_user_policy(PolicyDocument=policy_document_allow,
+                                      PolicyName='DenyGetPolicy', UserName=get_alt_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    # Copy object user1buck/srcobj to user2buck/dstobj  as user2
+    copy_source = {"Bucket": user1buck, 'Key': 'srcobj'}
+    e = assert_raises(ClientError, s3_client_alt.copy_object, Bucket=user2buck,
+                      CopySource=copy_source, Key='dstobj')
+    status, error_code = _get_status_and_error_code(e.response)
+    eq(status, 403)
+    eq(error_code, 'AccessDenied')
+    response = client.delete_user_policy(PolicyName='DenyGetPolicy', UserName=get_alt_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+
+def test_copy_object_allow_get_src_put_dest():
+    client = get_iam_client()
+    # Create bucket user1buck and upload object srcobj as user1
+    s3_client_iam = get_iam_s3client()
+    user1buck = get_new_bucket(client=s3_client_iam)
+    response = s3_client_iam.put_object(Bucket=user1buck, Key='srcobj', Body='bar')
+    etag = response['ETag']
+    # Create bucket user2buck as user2
+    s3_client_alt = get_alt_client()
+    user2buck = get_new_bucket(client=s3_client_alt)
+    # Add inline user policy for user3 with allow get on source and put on destination
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": [
+             {"Effect": "Allow",
+              "Action": ["s3:GetObject"],
+              "Resource": f"arn:aws:s3:::{user1buck}/*"},
+             {"Effect": "Allow",
+              "Action": ["s3:PutObject"],
+              "Resource": f"arn:aws:s3:::{user2buck}/*"}
+         ]}
+    )
+    response = client.put_user_policy(PolicyDocument=policy_document_allow,
+                                      PolicyName='AllowPolicy', UserName=get_main_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    # Copy object user1buck/srcobj to user2buck/dstobj as user3
+    main_s3_client = get_client()
+    copy_source = {"Bucket": user1buck, 'Key': 'srcobj'}
+    main_s3_client.copy_object(Bucket=user2buck, CopySource=copy_source, Key='dstobj')
+    # GET Object user2buck/dstobj as user2
```

Comment:
in comment its mentioned , as user2, but you are performing that operation on 3rd user. pls modify the comment  and in tds as well if that (user3)is expected

Reply:
we have not permitted user3 for get object on bucket. Pls check and modify appropriately, which user needs to perform get on user2buck/dstobj

Reply:
The same policy has two statements to allow Put on user2buck and allow get on user1buck

Reply:
Thanks for correcting TDS